src/main/kotlin/org/rust/ide/utils/import/ImportCandidatesCollector.kt

```diff
@@ -350,7 +354,22 @@ sealed class ImportInfo {
     }
 }
 
-data class ImportCandidate(val qualifiedNamedItem: QualifiedNamedItem, val info: ImportInfo)
+data class ImportCandidate(val qualifiedNamedItem: QualifiedNamedItem, val info: ImportInfo) : Comparable<ImportCandidate> {
+    override fun compareTo(other: ImportCandidate): Int {
+        val origin = qualifiedNamedItem.containingCrate?.origin
+        val otherOrigin = other.qualifiedNamedItem.containingCrate?.origin
+        val userPath = info.usePath
+        val otherUsePath = info.usePath
```

Comment:
```suggestion
        val otherUsePath = other.info.usePath
```
It's a reason of weird candidates order in tests